activestorage/active.py

```diff
@@ -406,22 +448,39 @@ def _process_chunk(self, session, fsref, chunk_coords, chunk_selection, counts,
         key = f"{self.ncvar}/{coord}"
         rfile, offset, size = tuple(fsref[key])
 
+        # S3: pass in pre-configured storage options (credentials)
         if self.storage_type == "s3":
             parsed_url = urllib.parse.urlparse(rfile)
             bucket = parsed_url.netloc
             object = parsed_url.path
-            # FIXME: We do not get the correct byte order on the Zarr Array's dtype
-            # when using S3, so use the value captured earlier.
-            dtype = self._dtype
-            tmp, count = reductionist.reduce_chunk(session, S3_ACTIVE_STORAGE_URL,
-                                                   S3_URL,
-                                                   bucket, object, offset,
-                                                   size, compressor, filters,
-                                                   missing, dtype,
-                                                   self.zds._chunks,
-                                                   self.zds._order,
-                                                   chunk_selection,
-                                                   operation=self._method)
+            if not self.storage_options:
+                # FIXME: We do not get the correct byte order on the Zarr Array's dtype
+                # when using S3, so use the value captured earlier.
+                dtype = self._dtype
+                tmp, count = reductionist.reduce_chunk(session,
+                                                       S3_ACTIVE_STORAGE_URL,
+                                                       S3_URL,
+                                                       bucket, object, offset,
+                                                       size, compressor, filters,
+                                                       missing, dtype,
+                                                       self.zds._chunks,
+                                                       self.zds._order,
+                                                       chunk_selection,
+                                                       operation=self._method)
+            else:
```

Comment:
Why do we need a special case for `if not self.storage_options:`? See previous comment on pre-configured options ...

Reply:
hey bud @davidhassell - also RE your comment above: very good question and counter-intuitive at first glance indeed, I agree. But here's my logic behind it:
- let's keep a hardwired configuration for now, test the StackHPC MinIO implemented functionality continually - so far, it's the only S3 configuration that actually works
- when we have tested with the `storage_options` and are sure of its functionality and generality, my intention is still to keep that configuration that we have now, but make it available through not reading an `activestorage/config.py` module but rather a number of set piece configuration yaml files that we have pre-configured for various sites - organizations can contribute those once they have S3 object stores

Let's talk about it tomorrow, shall we?